src/plcc/parse_spec/parse_syntactic_spec/parse_syntactic_spec.py

```diff
@@ -0,0 +1,137 @@
+from typing import List
+from plcc.load_spec.load_rough_spec.parse_dividers import Divider
+from plcc.load_spec.load_rough_spec.parse_lines import Line
+import re
+from re import Match
+from .structs import (
+    SyntacticSpec,
+    SyntacticRule,
+    Symbol,
+    CapturingTerminal,
+    RepeatingSyntacticRule,
+    StandardSyntacticRule,
+    LhsNonTerminal,
+    RhsNonTerminal,
+    Terminal,
+    CapturingSymbol,
+    MalformedLHSError,
+    MalformedBNFError,
+)
+
+
+def parse_syntactic_spec(lines: List[Line | Divider] | None) -> SyntacticSpec:
+    return SyntacticParser(lines).parseSpec()
+
+
+class SyntacticParser:
+    def __init__(self, input: List[Line | Divider] | None):
+        self.spec = SyntacticSpec()
+        self.lines = input
+
+    def parseSpec(self) -> SyntacticSpec:
+        if not self.lines:
+            return self.spec
+        for line in self.lines[1:]:
+            parser = SyntacticParserHelper(line)
+            if not parser._isSkippable():
+                self.spec.append(parser.parseSyntacticRule())
+        return self.spec
+
+
+class SyntacticParserHelper:
```

Comment:
Let's improve the name...

This class is responsible for parsing a line in the syntactic section. So how about SyntacticLineParser?